server/routerlicious/packages/services-core/src/mongoDocumentRepository.ts

```diff
@@ -0,0 +1,44 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { FindAndOperateDocumentMode, ICollection, IDb, IDocumentRepository } from "./database";
+import { IDocument } from "./document";
+
+export class MongoDocumentRepository implements IDocumentRepository {
+    constructor(private readonly collection: ICollection<IDocument>) {
+    }
+
+    public static create(
+        database: IDb,
+        collectionName: string,
+    ): MongoDocumentRepository {
+        return new MongoDocumentRepository(database.collection<IDocument>(collectionName));
+    }
+
+    async readDocument(filter: any): Promise<IDocument> {
+        return this.collection.findOne(filter);
+    }
+
+    async updateDocument(
+        filter: any,
+        updates: any,
+        options?: any,
+    ): Promise<void> {
+        await (options.upsert
+            ? this.collection.upsert(filter, updates, options?.addToSet)
+            : this.collection.update(filter, updates, options?.addToSet));
+    }
+
+    async findAndOperateDocument(filter: any, updates: any, options: any): Promise<{ value: IDocument; existing: boolean; }> {
```

Comment:
>findAndOperateDocument [](http://example.com/codeflow?start=10&length=22)

Why not just directly surface findAndUpdate() and findOrCreate()?

Reply:
Done